kinto_client/__init__.py

```diff
@@ -199,10 +199,25 @@ def _get_cache_headers(self, safe, data=None, last_modified=None):
             return {'If-Match': utils.quote(last_modified)}
         # else return None
 
+    def _get_or_create(self, method_name, *args, **kwargs):
+        try:
+            method = getattr(self, method_name)
+            return method(*args, **kwargs)
+        except KintoException as e:
+            if not hasattr(e, 'response') or e.response.status_code != 412:
+                raise e
+
     # Buckets
 
     def create_bucket(self, bucket=None, data=None, permissions=None,
-                      safe=True):
+                      safe=True, if_not_exists=False):
+        if if_not_exists:
+            return self._get_or_create(
+                'create_bucket',
+                bucket=bucket,
+                data=data,
+                permissions=permissions,
+                safe=safe)
```

Comment:
Then I might change them all to the former, because I tend to prefer the one with everything at the next line (works in all cases)